spec/lib/tasks_helpers/metrics_spec.rb

```diff
@@ -0,0 +1,102 @@
+describe 'Metrics' do
+  let(:header) do
+    %w[
+      tenant
+      portfolio_count
+      product_count
+      portfolio_share_count
+    ].collect(&:titleize)
+  end
+
+  let(:tenant) do
+    FactoryBot.create(
+      :tenant,
+      :name            => 'tenant_portfolio_shared_context',
+      :external_tenant => 'external_tenant_portfolio_shared'
+    )
+  end
+
+  def create_portfolio
+    portfolio = FactoryBot.create(:portfolio)
+    FactoryBot.create(:access_control_entry, :aceable => portfolio)
+    portfolio.update_metadata
+  end
+
+  def create_shared_portfolio
+    portfolio = FactoryBot.create(:portfolio)
+    FactoryBot.create(
+      :access_control_entry,
+      :has_read_and_update_permission,
+      :aceable => portfolio
+    )
+    portfolio.update_metadata
+  end
+
+  def create_portfolio_for_tenant
+    ActsAsTenant.with_tenant(tenant) { create_portfolio }
+  end
+
+  def create_shared_portfolio_for_tenant
+    ActsAsTenant.with_tenant(tenant) { create_shared_portfolio }
+  end
+
+  describe '`.portfolio_count`' do
+    it 'returns metric' do
+      FactoryBot.create(:portfolio)
+
+      expect(TaskHelpers::Metrics.portfolio_count).to eq(1)
+    end
+  end
+
+  describe '`.number_of_products`' do
+    it 'returns metric' do
+      FactoryBot.create(:portfolio_item)
+
+      expect(TaskHelpers::Metrics.product_count).to eq(1)
+    end
+  end
+
+  describe '`.portfolio_share_count`' do
+    it 'returns metrics' do
+      create_portfolio
+      create_shared_portfolio
+
+      expect(TaskHelpers::Metrics.portfolio_share_count).to eq(1)
+    end
+  end
+
+  describe '`.per_tenant`' do
+    it 'returns only header for blank DB' do
+      expect(TaskHelpers::Metrics.per_tenant).to eq([header])
+    end
+
+    context '#with_data' do
+      let(:expected_result) do
+        [
+          header,
+          [Tenant.first.external_tenant, 2, 0, 1],
+          [tenant.external_tenant, 3, 0, 2]
+        ]
+      end
+
+      before do
+        create_portfolio
+        create_portfolio_for_tenant
+        create_shared_portfolio
+        create_shared_portfolio_for_tenant
+        create_shared_portfolio_for_tenant
+      end
+
+      it 'returns metrics' do
+        expect(TaskHelpers::Metrics.per_tenant).to eq(expected_result)
+      end
+
+      it 'returns metrics without empty tenants' do
+        FactoryBot.create(:tenant, :external_tenant => 'empty_tenant')
```

Comment:
This should be `create(:tenant, :external_tenant => 'empty_tenant')`, but also it should be placed in a context with a description similar to the `it` block, while the `it` block description changes to assert what you are testing.